Source/Core/DolphinQt/Config/Mapping/MappingCommon.cpp

```diff
@@ -83,14 +83,24 @@ QString DetectExpression(QPushButton* button, ciface::Core::DeviceContainer& dev
 
   button->setText(old_text);
 
-  if (!input)
-    return {};
+  QString full_expression;
 
-  ciface::Core::DeviceQualifier device_qualifier;
-  device_qualifier.FromDevice(device.get());
+  for (auto [device, input] : detections)
+  {
+    ciface::Core::DeviceQualifier device_qualifier;
+    device_qualifier.FromDevice(device.get());
+
+    if (!full_expression.isEmpty())
+      full_expression += QChar::fromLatin1('+');
+
+    full_expression += MappingCommon::GetExpressionForControl(
+        QString::fromStdString(input->GetName()), device_qualifier, default_device, quote);
+  }
+
+  if (detections.size() > 1)
+    return QString::fromUtf8("@(") + std::move(full_expression) + QString::fromUtf8(")");
```

Comment:
```suggestion
    return QStringLiteral("@(%1)").arg(full_expression);
```

Is a slight bit more concise (and signals the strings are indeed intended to be non-translatable).